RecoParticleFlow/PFClusterProducer/plugins/CorrectedECALPFClusterProducer.cc

```diff
@@ -16,37 +16,26 @@
 
 #include "RecoParticleFlow/PFClusterProducer/interface/PFClusterEMEnergyCorrector.h"
 #include "RecoParticleFlow/PFClusterTools/interface/LinkByRecHit.h"
-#include "TVector2.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
 
 namespace {
   typedef reco::PFCluster::EEtoPSAssociation::value_type EEPSPair;
   bool sortByKey(const EEPSPair& a, const EEPSPair& b) {
     return a.first < b.first;
   } 
-  double testPreshowerDistance(const edm::Ptr<reco::PFCluster>& eeclus,
-			       const edm::Ptr<reco::PFCluster>& psclus) {
-    if( psclus.isNull() ) return -1.0;
-    /* 
-    // commented out since PFCluster::layer() uses a lot of CPU
-    // and since 
-    if( PFLayer::ECAL_ENDCAP != eeclus->layer() ) return -1.0;
-    if( PFLayer::PS1 != psclus->layer() &&
-	PFLayer::PS2 != psclus->layer()    ) {
-      throw cms::Exception("testPreshowerDistance")
-	<< "The second argument passed to this function was "
-	<< "not a preshower cluster!" << std::endl;
-    } 
-    */
-    const reco::PFCluster::REPPoint& pspos = psclus->positionREP();
-    const reco::PFCluster::REPPoint& eepos = eeclus->positionREP();
+
+  // why, why -1 and not <double>::max() ????
+  double testPreshowerDistance(reco::PFCluster const & eeclus,
+			       reco::PFCluster const & psclus) {
+    auto const & pspos = psclus.positionREP();
+    auto const & eepos = eeclus.positionREP();
     // lazy continue based on geometry
-    if( eeclus->z()*psclus->z() < 0 ) return -1.0;
-    const double dphi= std::abs(TVector2::Phi_mpi_pi(eepos.phi() - 
-						     pspos.phi()));
+    if( eeclus.z()*psclus.z() < 0 ) return -1.0;
+    auto deta= std::abs(eepos.eta() - pspos.eta());
+    if( deta > 0.3 ) return -1.0;
+    auto dphi= deltaPhi(eepos.phi(),  pspos.phi());
```

Comment:
missing abs!
will fix next iteration, it is just slower as it calls LinkByRecHit... three time more